python/cudf/cudf/core/frame.py

```diff
@@ -2089,7 +2090,20 @@ def product(
         b    5040
         dtype: int64
         """
-        axis = self._get_axis_from_axis_arg(axis)
+        if axis in {no_default, None}:
+            if axis is None:
+                # Do not remove until pandas 3.0 support is added.
+                warnings.warn(
```

Comment:
Why doesn't this run through the same logic as the other `_reduce` calls, and get the warning from `_reduce`?

Could we instead replace the logic below, and write the cupy dispatch as `"product" if (axis == 0 or axis is None) else "prod"`?

Reply:
Yup, we can rely on `_reduce`. I also simplified the `prod` vs `product` logic selection.